udev_monitor.c

```diff
@@ -1,36 +1,30 @@
+#include <poll.h>
 #include <fcntl.h>
 #include <stdio.h>
-#include <signal.h>
 #include <string.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <limits.h>
 #include <pthread.h>
 #include <sys/stat.h>
-#include <sys/epoll.h>
 #include <sys/socket.h>
 #include <sys/inotify.h>
 
 #include "udev.h"
 #include "udev_list.h"
 
-#define THREADS_MAX 5
-
 #ifndef UDEV_MONITOR_DIR
 #define UDEV_MONITOR_DIR "/tmp/.libudev-zero"
 #endif
 
 struct udev_monitor {
     struct udev_list_entry subsystem_match;
     struct udev_list_entry devtype_match;
-    pthread_t thread[THREADS_MAX];
-    pthread_barrier_t barrier;
+    pthread_t thread;
```

Comment:
Move this to `udev_monitor_enable_receiving()`

Reply:
So there was one aspect I was hesitating over. We need to wait for the thread to finish before freeing the udev_monitor. I had made the thread joinable and after it was woken up waited on it. This is why I kept it has a field instead of making it a local var.